packages/app-desktop/gui/NoteEditor/utils/contextMenuUtils.ts

```diff
@@ -0,0 +1,53 @@
+import Resource from '@joplin/lib/models/Resource';
+
+export enum ContextMenuItemType {
+	None = '',
+	Image = 'image',
+	Resource = 'resource',
+	Text = 'text',
+	Link = 'link',
+}
+
+export interface ContextMenuOptions {
+	itemType: ContextMenuItemType;
+	resourceId: string;
+	tempResource: {
+		data: any; // string or Uint8Array
+		mime: string;
+		filename: string;
+	};
```

Comment:
Why is it called "tempResource"? Please could you describe how it's used in your code?

Reply:
Previously I used `resourceContent` instead to pass the data. I renamed it `tempResource` since it now approximately describes a resource that is temporary, i.e not actually present in the resource db or in the local filesystem. It was generated on the fly. Using this info we can generate a new local file or copy the contents or generate a data uri as per our needs.

Reply:
Thanks for clarifying, I understand why you've done it that way now. But to be consistent with the rest of the interface please put the data in the existing "textToCopy" then add a mime and filename prop, and remove the tempResource key. There's no need for a nested structure here.